### PR TITLE
Sharded pubsub command execution within multi/exec

### PR DESCRIPTION
**Pull request by hpatro**
Allow SPUBLISH command within multi/exec on replica

Behavior on unstable:

```
127.0.0.1:6380> CLUSTER NODES
39ce8aa20f1f0d91f1a88d976ee1926dfefcdf1a 127.0.0.1:6380@16380 myself,slave 8b0feb120b68aac489d6a5af9c77dc40d71bc792 0 0 0 connected
8b0feb120b68aac489d6a5af9c77dc40d71bc792 127.0.0.1:6379@16379 master - 0 1705091681202 0 connected 0-16383
127.0.0.1:6380> SPUBLISH hello world
(integer) 0
127.0.0.1:6380> MULTI
OK
127.0.0.1:6380(TX)> SPUBLISH hello world
QUEUED
127.0.0.1:6380(TX)> EXEC
(error) MOVED 866 127.0.0.1:6379
```

With this change:

```
127.0.0.1:6380> SPUBLISH hello world
(integer) 0
127.0.0.1:6380> MULTI
OK
127.0.0.1:6380(TX)> SPUBLISH hello world
QUEUED
127.0.0.1:6380(TX)> EXEC
1) (integer) 0
```

### CODE REVIEW

File: tests/unit/cluster/sharded-pubsub.tcl [hunk 1]
```diff
@@ -0,0 +1,56 @@
+start_cluster 1 1 {tags {external:skip cluster}} {
+        set primary_id 0
+        set replica1_id 1
+
+        set primary [Rn $primary_id]
+        set replica [Rn $replica1_id]
```
Comment (madolson) [inline tests/unit/cluster/sharded-pubsub.tcl]: Spacing seems wonked now.

File: src/cluster.c [hunk 2]
```diff
@@ -1030,9 +1030,11 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
...
     }
 
-    int is_pubsubshard = cmd->proc == ssubscribeCommand ||
-                         cmd->proc == sunsubscribeCommand ||
-                         cmd->proc == spublishCommand;
+    uint64_t cmd_flags = getCommandFlags(c);
+    
+    /* Only valid for sharded pubsub as regular pubsub can operate on any node and bypasses this layer. */
+    int pubsubshard_included = (cmd_flags & CMD_PUBSUB) ||
+                           (c->cmd->proc == execCommand && (c->mstate.cmd_flags & CMD_PUBSUB));
```
Comment (zuiderkwast) [inline src/cluster.c:1037]: How does regular pubsub bypass this layer? Does regular publish work within multi-exec? Please remind us.

Comment (madolson) [reply src/cluster.c:1037]: pubsub is not a key based command, so it doesn't even enter the getNodeByQuery :)

Comment (zuiderkwast) [reply src/cluster.c:1037]: Ah, this is getNodeByQuery. Thanks for the context. :)

**Review by zuiderkwast — APPROVED** [review]

**Review by madolson — APPROVED** [review]

**Pull request merged**